lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -575,33 +581,47 @@ SdfPath getDstSdfPath(const Ufe::Path& ufePulledPath, const SdfPath& srcSdfPath,
 //------------------------------------------------------------------------------
 //
 UsdMayaPrimUpdaterSharedPtr createUpdater(
+    const Ufe::Path&                 ufePulledPath,
     const SdfLayerRefPtr&            srcLayer,
-    const SdfPath&                   primSpecPath,
+    const SdfPath&                   srcPath,
+    const SdfLayerRefPtr&            dstLayer,
+    const SdfPath&                   dstPath,
     const UsdMayaPrimUpdaterContext& context)
 {
-    // Get the primSpec from the src layer.
-    auto primSpec = srcLayer->GetPrimAtPath(primSpecPath);
+    // The root of the pulled hierarchy is crucial for determining push
```

Comment:
Having a stable updater that performs both the pull and the push is crucial to success.  Ensure we get the same updater by using the original pulled path to create the updater for both pull and push.